src/combat.ts

```diff
@@ -615,46 +619,90 @@ export class Macro extends StrictMacro {
+    return result;
+  } catch (e) {
+    throw `Combat exception! Last macro error: ${get("lastMacroError")}. Exception ${e}.`;
+  }
+}
+
 /**
  * Attempt to perform a nonstandard combat-starting Action with a Macro
  * @param macro The Macro to attempt to use
  * @param action The combat-starting action to attempt
  * @param tryAuto Whether or not we should try to resolve the combat with an autoattack; autoattack macros can fail against special monsters, and thus we have to submit a macro with Macro.save() regardless
  * @returns The output of your specified action function (typically void)
  */
-export function withMacro<T>(macro: Macro, action: () => T, tryAuto = false): T {
+export function withMacro<T, M extends StrictMacro>(macro: M, action: () => T, tryAuto = false): T {
   if (getAutoAttack() !== 0) setAutoAttack(0);
   if (tryAuto) macro.setAutoAttack();
-  macro.save();
+  makeCcs(macro);
   try {
     return action();
   } finally {
-    Macro.clearSaved();
     if (tryAuto) setAutoAttack(0);
   }
 }
 
-export function main(): void {
-  if (have($effect`Eldritch Attunement`)) {
-    Macro.if_($monster`Eldritch Tentacle`, Macro.basicCombat())
-      .step(Macro.load())
-      .submit();
-  } else if (getTodaysHolidayWanderers().length !== 0) {
-    Macro.ifHolidayWanderer(
-      Macro.externalIf(
-        haveEquipped($item`backup camera`) &&
-          get("_backUpUses") < 11 &&
-          get("lastCopyableMonster") === $monster`Knob Goblin Embezzler` &&
-          myFamiliar() === meatFamiliar(),
-        Macro.skill($skill`Back-Up to your Last Enemy`).step(Macro.load()),
-        Macro.basicCombat()
-      )
-    )
-      .step(Macro.load())
-      .submit();
-  } else {
-    Macro.load().submit();
-  }
-  while (inMultiFight()) runCombat();
-  if (choiceFollowsFight()) visitUrl("choice.php");
+/**
+ * Adventure in a location and handle all combats with a given macro.
+ * To use this function you will need to create a consult script that runs Macro.load().submit() and a CCS that calls that consult script.
+ * See examples/consult.ts for an example.
+ *
+ * @category Combat
+ * @param loc Location to adventure in.
+ * @param macro Macro to execute.
+ */
+export function garboAdventure<M extends StrictMacro>(loc: Location, macro: M): void {
+  if (getAutoAttack() !== 0) setAutoAttack(0);
+  makeCcs(macro);
+  completeCombat(() => adv1(loc, -1, ""));
+}
+
+/**
+ * Adventure in a location and handle all combats with a given autoattack and manual macro.
+ * To use the nextMacro parameter you will need to create a consult script that runs Macro.load().submit() and a CCS that calls that consult script.
+ * See examples/consult.ts for an example.
+ *
+ * @category Combat
+ * @param loc Location to adventure in.
+ * @param autoMacro Macro to execute via KoL autoattack.
+ * @param nextMacro Macro to execute manually after autoattack completes.
+ */
+export function garboAdventureAuto<M extends StrictMacro>(
```

Comment:
Do we ever want to use something other than a Garbo macro? Is there a benefit to genericity here?

Reply:
There is a compelling reason that I had two days ago and now I have no recollection of what it is.